src/apiserver/api_server_impl.cc

```diff
@@ -83,6 +83,67 @@ void APIServerImpl::Process(google::protobuf::RpcController* cntl_base, const Ht
     cntl->response_attachment().append(writer.GetString());
 }
 
+class ExecContext {
+ public:
+    bool is_online = false;
+    bool is_sync = true;
+    int job_timeout = 600000;  // ms
+    ExecContext() = default;
+    ExecContext(bool online, bool sync) : is_online(online), is_sync(sync) {}
+
+    std::string ToString() const {
+        std::stringstream ss;
+        ss << (is_online ? "online" : "offline") << "-" << (is_sync ? "sync" : "async") << "-" << job_timeout << "ms";
+        return ss.str();
+    }
+};
+
+std::map<std::string, ExecContext> mode_map{
+    {"offsync", {false, true}}, {"offasync", {false, false}}, {"online", {true, false}}};
+
+void APIServerImpl::RegisterQuery() {
+    provider_.post("/dbs/:db_name",
+                   [this](const InterfaceProvider::Params& param, const butil::IOBuf& req_body, JsonWriter& writer) {
+                       auto resp = GeneralResp();
+                       auto db_it = param.find("db_name");
+                       if (db_it == param.end()) {
+                           writer << resp.Set("url has no db_name");
+                           return;
+                       }
+                       auto db = db_it->second;
+
+                       // default mode is offsync
+                       QueryReq req;
+                       JsonReader query_reader(req_body.to_string().c_str());
+                       query_reader >> req;
+                       if (!query_reader) {
+                           writer << resp.Set("Json parse failed, " + req_body.to_string());
+                           return;
+                       }
+                       auto mode = boost::to_lower_copy(req.mode);
+                       auto it = mode_map.find(mode);
+                       if (it == mode_map.end()) {
+                           writer << resp.Set("Invalid mode " + mode);
+                           return;
+                       }
+                       ExecContext ctx = it->second;
+
+                       const auto& sql = req.sql;
+                       LOG(INFO) << "post [" << ctx.ToString() << "] query on db [" << db << "], sql: " << sql;
+                       // TODO(hw): if api server supports standalone, we should check if cluster mode here
+
+                       hybridse::sdk::Status status;
+                       // TODO(hw): if sql is not a query, it may be a ddl, we use ExecuteSQL to execute it before we
+                       //  supports ddl http api. It's useful for api server tests(We can create table when we only
+                       //  connect to the api server).
```

Comment:
support executing ddl or insert? #2002

Reply:
how to get the running status  if execute an offline async job

Reply:
Yep. Maybe issue 2002 can be 'write the unit tests for all sql types'?